src/components/ArticleManager.jsx

```diff
@@ -5,23 +5,84 @@ import { useSearchParams } from 'react-router-dom';
 
 export default function ArticleManager(){
     const [articleList, setArticleList] = useState([])
+    const [isLoading, setIsloading] = useState(true)
 
     const [searchParams, setSearchParams] = useSearchParams();
     const topicQuery = searchParams.get('topic');
     const authorQuery = searchParams.get('author')
+
+    const [direction, setDirection] = useState('desc')
+    const [sortField, setSortField] = useState("null")
+
+    console.log(sortField)
+    // Setting direction asc/deesc
+    const onDirectionChange = (e) =>{
+        setDirection(e.target.value)
+    }
+    const setDirectionOrder = (e) => {
+        onDirectionChange(e)
+        const newParams = new URLSearchParams(searchParams);
+        newParams.set('order', direction);
+        setSearchParams(newParams)
+    }
+
+    // sorting 
+    const sortBasedOnKey = (property) =>{
```

Comment:
You are providing this information to the API request (as queries) which should sort them, so we shouldn't need to sort it again if a new API request is triggered 🤔